tests/x11/yast2_lan_restart.pm

```diff
@@ -42,7 +42,20 @@ sub check_network {
     $status //= 'no_restart';
     wait_still_screen;
     send_key 'alt-o';                     # OK
-    assert_screen 'yast2_closed_xterm_visible', 120;
+                                          # new: warning pops up for firewall, alt-y for assign it to zone
+    assert_screen([qw(yast2-lan-restart_firewall_active_warning yast2_closed_xterm_visible)], 120);
+    if (match_has_tag 'yast2-lan-restart_firewall_active_warning') {
+        send_key 'alt-y';
+        wait_still_screen;
+        send_key 'alt-n';
+        wait_still_screen;
+        send_key 'alt-o';
+    }
+    else {
+        assert_screen 'yast2_closed_xterm_visible', 120;
```

Comment:
It's not really a comment on the code but more a question: is `assert_screen 'yast2_closed_xterm_visible', 120;` really useful?
As if the this code is executed it should has been already did with `assert_screen ([qw(yast2-lan-restart_firewall_active_warning yast2_closed_xterm_visible)], 120);`, no?

And so it could be executed two times. But maybe I'm wrong :)

Reply:
I don't know. I just try to fix the code here ;)
Actually we want to re-work the code according to changes from sles 15/leap 15.0.
yast2_closed_xterm_visible', 120; -- is needed here so we can be sure that lan got closed correctly and it is ready for following tests. 

assert_screen checks only needles and has nothing to do with closing xterm.

Reply:
Could you try to remove the `else { assert_screen ... }` and test if it still works?

Reply:
well, yast2_closed_xterm_visible is needed anyway, for sles 12 and leap lower than 15.0
Just want to avoid to use check_screen here like if (check_screen 'yast2-lan-restart_firewall_active_warning') {...}

Reply:
Agreed that it is needed for older version, but `assert_screen([qw(yast2-lan-restart_firewall_active_warning yast2_closed_xterm_visible)], 120);` is not intended to test both? As I understand the code, it tests if one of `yast2-lan-restart_firewall_active_warning` or `yast2_closed_xterm_visible` is found. If `yast2-lan-restart_firewall_active_warning` is found it does specific stuff, and it it's the other it does assert_screen again, which may not be needed for me, as it has already been checked (on all sle/opensuse version).

But again, it's more for me to understand better how openQA works :-) This is why I asked if a test without it was possible or not.

Reply:
it is needed for both cases. so I added this yast2_closed_xterm_visible now. It looks better:
http://e13.suse.de/tests/26#step/yast2_lan_restart/39